### PR TITLE
gd32e10x: update hal to v1.3.0

### PR DESCRIPTION
**Pull request by feilongfl**
update gd32e10x hal to latest version.

hal download from:
https://www.gd32mcu.com/en/download/7?kw=GD32E1#:~:text=GD32E10x%20Firmware%20Library

### CODE REVIEW

**Comment by nandojve** [comment]
Hi @feilongfl ,

Don't forget to open the Zephyr PR that updates west.yaml with git hash.

**Comment by feilongfl** [comment]
> Hi @feilongfl ,
> 
> Don't forget to open the Zephyr PR that updates west.yaml with git hash.

update zephyr west with PR https://github.com/zephyrproject-rtos/zephyr/pull/49297

File: gd32e10x/standard_peripheral/include/gd32e10x_i2c.h [hunk 1]
```diff
@@ -309,6 +309,11 @@ typedef enum {
 #define I2C_ADDFORMAT_7BITS           ((uint32_t)0x00000000U)                  /*!< address format is 7 bits */
 #define I2C_ADDFORMAT_10BITS          I2C_SADDR0_ADDFORMAT                     /*!< address format is 10 bits */
 
+#define I2CCLK_MAX ((uint32_t)0x0000003FU)/*!< i2cclk maximum value */
+#define I2CCLK_MIN ((uint32_t)0x00000002U)/*!< i2cclk minimum value for standard mode */
+#define I2CCLK_FM_MIN ((uint32_t)0x00000008U)/*!< i2cclk minimum value for fast mode */
+#define I2CCLK_FM_PLUS_MIN ((uint32_t)0x00000018U)/*!< i2cclk minimum value for fast mode plus */                     /*!< address format is 10 bits */
```
Comment (gmarull) [inline gd32e10x/standard_peripheral/include/gd32e10x_i2c.h:315]: trailing comment seems wrong here?

**Review by gmarull — APPROVED** [review]

File: gd32e10x/standard_peripheral/include/gd32e10x_i2c.h (hunk 1 above)
Comment (feilongfl) [reply gd32e10x/standard_peripheral/include/gd32e10x_i2c.h]: You are right, I made a mistake here.

**Pull request merged**